artifact/gem5art/artifact/artifact.py

```diff
@@ -260,6 +286,27 @@ def __init__(self, other: Union[str, UUID, Dict[str, Any]]) -> None:
         self.cwd = Path(other['cwd'])
         self.inputs = [Artifact(i) for i in other['inputs']]
 
+        # Optional fields
+        self.architecture = other['architecture'] if 'architecture' in other else ''
```

Comment:
I think this whole thing can be simplified by doing the following
```suggestion
        self.architecture = other.get('architecture')
```

Note, you can use a second parameter to `get` to set a default. None probably works, though. See https://docs.python.org/3/library/stdtypes.html#dict.get